File: project_euler/Problem075/Problem075.java
package project_euler;

import java.util.HashMap;
import java.util.Map;

public class Problem075 {
    public void solution() {
        int target = 1500000;
        int border = (int) Math.sqrt(target >> 1);
        Map<Integer, Integer> perimeters = new HashMap<>();
        for (int i = 2; i <= border; ++i) {
            for (int j = 1; j < i; ++j) {
                if (0 != ((i + j) & 1) && isRelativelyPrime(i, j)) {
                    int pivot = i * (i + j) << 1;
                    int length = pivot;
                    while (length < target) {
                        perimeters.put(length, perimeters.getOrDefault(length, 0) + 1);
                        length += pivot;
                    }
                }
            }
        }

        int result = 0;
        for (Map.Entry<Integer, Integer> entry : perimeters.entrySet()) {
            if (1 == entry.getValue()) {
                ++result;
            }
        }
        System.out.println(result);
    }

    private boolean isRelativelyPrime(int i, int j) {
        while (0 != i % j) {
            int auxiliary = i % j;
            i = j;
            j = auxiliary;
        }
        return 1 == j;
    }
}

File: project_euler/Problem075/Problem075.js
function Problem075() {
    let target = 1500000
    let border = Math.floor(Math.sqrt(target >>> 1))
    let perimeters = new Map()
    for (let i = 2; i <= border; ++i) {
        for (let j = 1; j < i; ++j) {
            if (0 !== ((i + j) & 1) && isRelativelyPrime(i, j)) {
                let pivot = i * (i + j) << 1
                let length = pivot;
                while (length < target) {
                    if (perimeters.has(length)) {
                        perimeters.set(length, perimeters.get(length) + 1)
                    } else {
                        perimeters.set(length, 1)
                    }
                    length += pivot;
                }
            }
        }
    }

    let result = 0
    for (let [, value] of perimeters) {
        if (1 === value) {
            ++result
        }
    }
    console.log(result)
}

function isRelativelyPrime(i, j) {
    while (i % j) {
        let auxiliary = i % j
        i = j
        j = auxiliary
    }
    return 1 === j
}

module.exports = {
    Problem075 : Problem075
};
